summarization/text_summarization.py

```diff
@@ -51,4 +61,53 @@ def summarize_text(self, text: str, ratio=0.4):
             raise IncorrectInputError(
                 "Ratio should be in the range of [0,1] ")
 
-        return self.text_summarizer(text, ratio)
+        self.text = text
+
+        # get the entities in the text and store it
+        # in a list
+        self._get_entites_from_text()
+        sentence_tokenized_text = sent_tokenize(text)
+
+        text_with_entities_list = list()
+
+        # separate sentences into two:
+        # those that contain entities
+        # and those that don't
+        for sentence in sentence_tokenized_text:
+            sentence_has_entity = False
+
+            for entity in self.entity_list:
+                if entity in sentence:
+                    sentence_has_entity = True
+
+            if sentence_has_entity:
+                text_with_entities_list.append(sentence)
+
+        # summarize the sentence that don't have entities
+        tokenized_summarized_text \
+            = sent_tokenize(
+                self.text_summarizer(text, ratio)
+            )
+
+        # combines both enetity and non-entity sentences
+        # in ORDER
+        # order is necessary so the indexing
+        # is preserved
+        combined_summarized_text_list = list()
+
+        # iterate over the un-summarized sentences
+        # if the sentence has an entity or is in
+        # the summarized sentences list add it to
+        # the combined summarized list
+
+        for sentence in sentence_tokenized_text:
+            sentence_has_entity = False
+
+            for entity in self.entity_list:
+                if entity in sentence:
```

Comment:
Will it be possible to re-use `text_with_entities_list` here?

Reply:
Thanks for pointing this out. Updated accordingly.